secret.go

```diff
@@ -0,0 +1,53 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	_ interface {
+		json.Marshaler
+		yaml.Marshaler
+		selfInterpolater
+	} = (*Secret)(nil)
+)
+
+// Secret represents a pipeline secret configuration.
+type Secret struct {
+	Key                 string         `json:"key" yaml:"key"`
+	EnvironmentVariable *string        `json:"environment_variable,omitempty" yaml:"environment_variable,omitempty"`
```

Comment:
I wonder if this should be made `string` instead of `*string`? 

The empty string is treated as "empty" for the purposes of `omitempty`. The empty string is also definitely _not_ a valid environment variable name, so seems to me that there's no benefit to distinguishing between "not provided" and "empty but provided".

```suggestion
	EnvironmentVariable string        `json:"environment_variable,omitempty" yaml:"environment_variable,omitempty"`
```

Reply:
So I'd like to be able to support a future where we extend secrets with other types (ssh keys, files etc), and by having the environment variable by nullable, we could return a type of `{ key: "SECRET_NAME", file: {...} }` from the API as `step`. If it's not nullable, we'd need to return `{ key: "SECRET_NAME", file: {...}, environment_variable: "" }` which while possible, doesn't seem accurate or within our normal API conventions.

Reply:
If an attribute is "missing" from the JSON or YAML, then the unmarshaler won't return an error, it just won't fill in the field. So, the API wouldn't need to include `environment_variable: ""`. `omitempty` takes care of the other direction (skipping the attribute in the output when it is empty).

Reply:
> If an attribute is "missing" from the JSON or YAML, then the unmarshaler won't return an error, it just won't fill in the field.

Will that mean that `environment_variable` will be `nil` or "" when not set by the API?

Reply:
It would be left at the default value for the type - for `*string` that's `nil`, for `string` that is `""`.

Reply:
Thanks!